### DISCUSSION TITLE
Tree and Select with Material UI checkbox

### DISCUSSION THREAD

**Discussion by aubsparrow** [table-library/react-table-library#88] [General]
I am trying to create a table with select & tree and I'm using Material UI checkboxes. Are there any examples of how to do that? The examples in the documentation only work without a tree. 

The example will select the parent but not the children. 
![image](https://user-images.githubusercontent.com/50328837/203132374-003b73f2-3c3b-4e18-b923-8ec45f630bb3.png)

**Comment by rwieruch** [comment]
Start with this one https://react-table-library.com/?path=/docs/mixing-features-tree-select--base and exchange the default select checkbox with MUI checkbox which is demonstrated over here https://react-table-library.com/?path=/docs/features-select--custom-checkbox-material-ui Does this help?

**Reply by aubsparrow** [reply]
The problem I am running into with the provided material ui cusotm checkbox is how to make the parent rows in the tree display proper indeterminate state and check all the child rows when checked. Does that make sense?
![image](https://user-images.githubusercontent.com/50328837/203392584-dff168f7-565f-4999-bb8f-d9fac0bbe90c.png)

What is the exact structure of select.state?

**Comment by aubsparrow** [comment]
This is what I ended up having to put in the onChange method: 
`onChange={() => {
                            console.log(item);
                            if (item.nodes?.length !== undefined) {
                              if (item.nodes?.length > 0) {
                                const ids = new Array<string>();
                                ids.push(item.id);
                                for (const sat of item.nodes) {
                                  ids.push(sat.toString());
                                }
                                if (select.state.ids.includes(item.id)) {
                                  select.fns.onRemoveByIds(ids);
                                } else {
                                  select.fns.onAddByIds(ids, {});
                                }
                              }
                            } else {
                              select.fns.onToggleById(item.id);
                            }`
